opsgenie/resource_opsgenie_alert_policy_test.go

```diff
@@ -0,0 +1,178 @@
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("Checking all the teams to get our TeamID")
+	for _, u := range resp.Teams {
+		fmt.Printf("checking team: %s", u.Name)
+		if strings.HasPrefix(u.Name, "genieteam") {
+			log.Printf("Destroying alert policy for team %s", u.Name)
+			client2, err := policy.NewClient(meta.(*OpsgenieClient).client.Config)
+			if err != nil {
+				return err
+			}
+
+			resp2, err := client2.ListAlertPolicies(context.Background(), &policy.ListAlertPoliciesRequest{
+				TeamId: u.Id,
+			})
+			if err != nil {
+				return err
+			}
+			for _, k := range resp2.Policies {
+
+				deleteRequest := policy.DeletePolicyRequest{
+					Type:   policy.PolicyType(k.Type),
+					TeamId: u.Id,
+					Id:     k.Id,
+				}
+
+				fmt.Printf("Deleting policy %s", k.Name)
+				if _, err := client2.DeletePolicy(context.Background(), &deleteRequest); err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func TestAccOpsGenieAlertPolicy_basic(t *testing.T) {
+	alertPolicyName := acctest.RandString(6)
+
+	config := testAccOpsGenieAlertPolicy_basic(alertPolicyName)
+
+	resource.Test(t, resource.TestCase{
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckOpsGenieAlertPolicyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckOpsGenieAlertPolicyExists("opsgenie_alert_policy.test"),
+				),
+			},
+		},
+	})
+}
+
+func testCheckOpsGenieAlertPolicyDestroy(s *terraform.State) error {
+	client, err := policy.NewClient(testAccProvider.Meta().(*OpsgenieClient).client.Config)
+	if err != nil {
+		return err
+	}
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "opsgenie_alert_policy" {
+			continue
+		}
+		req := policy.GetAlertPolicyRequest{
+			Id: rs.Primary.Attributes["id"],
+		}
+		_, err := client.GetAlertPolicy(context.Background(), &req)
+		if err != nil {
+			x := err.(*ogClient.ApiError)
+			if x.StatusCode != 404 {
+				return errors.New(fmt.Sprintf("Alert policy still exists : %s", x.Error()))
+			}
+		}
+	}
+
+	return nil
+}
+
+func testCheckOpsGenieAlertPolicyExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		fmt.Printf("Running testCheckOpsGenieAlertPolicyExists")
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		id := rs.Primary.Attributes["id"]
+
+		fmt.Printf("Got ID for policy: %s", name)
+		client, err := policy.NewClient(testAccProvider.Meta().(*OpsgenieClient).client.Config)
+		if err != nil {
+			return err
+		}
+		req := policy.GetAlertPolicyRequest{
+			TeamId: rs.Primary.Attributes["team_id"],
+			Id:     rs.Primary.Attributes["id"],
+		}
+
+		_, err = client.GetAlertPolicy(context.Background(), &req)
+		if err != nil {
+			return fmt.Errorf("Bad: Alert policy %q does not exist", id)
+		}
+		return nil
+	}
+}
+
+func testAccOpsGenieAlertPolicy_basic(alertPolicyName string) string {
```

Comment:
could you add another test function since this has a lot of fields we dont want to miss anything

Reply:
Added test cases with other feields